PWGEM/PhotonMeson/Utils/gammaSelectionCuts.h

```diff
@@ -0,0 +1,158 @@
+    } else {
+      registry.fill(HIST("hCaloCuts_EMC"), 2, iCut);
+    }
+  }
+  return cut_return;
+}
+
+uint64_t doMinECutEMC(int iCut, uint64_t cutbit, aod::SkimEMCCluster const& cluster, HistogramRegistry registry)
+{
+  uint64_t cut_return = 0;
+  if (cutbit & ((uint64_t)1 << (uint64_t)iCut)) {        // check if current cut should be applied
+    if (cluster.energy() > emccuts::EMC_minE.at(iCut)) { // check cut itself
+      cut_return |= (1 << iCut);                         // set bit of current cut to 1 for passing the cut
+    } else {
+      registry.fill(HIST("hCaloCuts_EMC"), 3, iCut);
+    }
+  }
+  return cut_return;
+}
+
+uint64_t doNCellCutEMC(int iCut, uint64_t cutbit, aod::SkimEMCCluster const& cluster, HistogramRegistry registry)
+{
+  uint64_t cut_return = 0;
+  if (cutbit & ((uint64_t)1 << (uint64_t)iCut)) {             // check if current cut should be applied
+    if (cluster.nCells() >= emccuts::EMC_minNCell.at(iCut)) { // check cut itself
+      cut_return |= (1 << iCut);                              // set bit of current cut to 1 for passing the cut
+    } else {
+      registry.fill(HIST("hCaloCuts_EMC"), 4, iCut);
+    }
+  }
+  return cut_return;
+}
+
+uint64_t doTrackMatchingEMC(int iCut, uint64_t cutbit, aod::SkimEMCCluster const& cluster, aod::SkimEMCMTs const& tracks, HistogramRegistry registry)
+{
+  uint64_t cut_return = 0;
+  double dEta, dPhi;
+  if (cutbit & ((uint64_t)1 << (uint64_t)iCut)) { // check if current cut should be applied
+    bool hasMatchedTrack_EMC = false;
+    for (const auto& track : tracks) {
+      dEta = track.tracketa() - cluster.eta();
+      dPhi = track.trackphi() - cluster.phi();
+      if ((fabs(dEta) <= emccuts::EMC_TM_Eta.at(iCut).at(0) + pow(track.trackpt() + emccuts::EMC_TM_Eta.at(iCut).at(1), emccuts::EMC_TM_Eta.at(iCut).at(2))) &&
+          (fabs(dPhi) <= emccuts::EMC_TM_Phi.at(iCut).at(0) + pow(track.trackpt() + emccuts::EMC_TM_Phi.at(iCut).at(1), emccuts::EMC_TM_Phi.at(iCut).at(2))) &&
+          cluster.energy() / track.trackp() < emccuts::EMC_Eoverp.at(iCut)) { // check cut itself
+        hasMatchedTrack_EMC = true;                                           // set bit of current cut to 1 for passing the cut
+      }
+    }
+    if (hasMatchedTrack_EMC) {
+      registry.fill(HIST("hCaloCuts_EMC"), 5, iCut);
+    } else {
+      cut_return |= (1 << iCut);
+    }
+  }
+  return cut_return;
+}
+
+uint64_t doPhotonCutsEMC(int iCut, uint64_t cutbit, aod::SkimEMCCluster const& cluster, aod::SkimEMCMTs const& tracks, Preslice<o2::aod::SkimEMCMTs> perEMCClusterMT, HistogramRegistry registry)
+{
+  uint64_t cut_return = 0;
+  auto tracksMatchedEMC = tracks.sliceBy(perEMCClusterMT, cluster.globalIndex());
+  cut_return = doTimeCutEMC(iCut, cutbit, cluster, registry);
+  cut_return &= doM02CutEMC(iCut, cut_return, cluster, registry);
```

Comment:
is this cut_return on purpose or should it be cutbit?
Feels like I don't get this function :D
I thought in the end, cut_return shall contain a 1 for each predefined cutSet,  that lets cluster pass

Reply:
Let me try to explain because it took me a while to get to this myself:
So in cutbit we have the bit flag of which cut settings we want to use. If a bit is 0 we don't want that setting and we don't need to look at it. In doTimeCutEMC we check if a cluster passes only the cut settings we set to 1 in cutbit. If the cluster passes this cut, the value from cutbit stored in cut_return will stay 1, if not it's turned to 0. So for the next cut, we no longer look for a cluster that failed the previous cut for a given cut setting. 
tl;dr: Thanks to using cut_return instead of cutbit after the timing cut we can prevent the other cuts from checking if the cluster survives that cut for a given cut setting if it failed to survive any of the previous cuts for that given cut setting.

Reply:
ahh I see, thanks! My 'problem' was, that I thought we want to know all the single cuts a cluster fails, not just one which it fails. 
Maybe explain that idea and logic in a comment? I guess it will save some time for people trying to understand it and start off with a wrong idea like I did

Reply:
For cluster QA this might be nice, but for us in the analysis it's not really interesting.
I can add a comment for the logic, sure!